robyn/reloader.py

```diff
@@ -0,0 +1,86 @@
+import os
+import signal
+import subprocess
+import sys
+import time
+import psutil
+
+from watchdog.events import FileSystemEventHandler
+from watchdog.observers import Observer
+
+from robyn.logger import Colors, logger
+
+
+def setup_reloader(directory_path: str, file_path: str):
+    event_handler = EventHandler(file_path)
+
+    event_handler.reload()
+
+    logger.info(
+        f"Dev server initialized with the directory_path : {directory_path}",
+        Colors.BLUE,
+    )
+
+    def terminating_signal_handler(_sig, _frame):
+        event_handler.stop_server()
+        logger.info("Terminating reloader", bold=True)
+        observer.stop()
+        observer.join()
+
+    signal.signal(signal.SIGINT, terminating_signal_handler)
+    signal.signal(signal.SIGTERM, terminating_signal_handler)
+
+    observer = Observer()
+    observer.schedule(event_handler, path=directory_path, recursive=True)
+    observer.start()
+
+    try:
+        while observer.is_alive():
+            observer.join(1)
+    finally:
+        observer.stop()
+        observer.join()
+
+
+class EventHandler(FileSystemEventHandler):
+    def __init__(self, file_path: str) -> None:
+        self.file_path = file_path
+
+        self.processes = []
```

Comment:
Thanks for the PR @AntoineRR 😄 

Do we still need this `self.processes` data member?

Reply:
Thanks, I forgot to remove it. I fixed it now :)